### ISSUE TITLE
NPE for mvn evosuite:coverage on root project, that contains no own tests but only modules

### ISSUE THREAD

**Issue by bseiller** [EvoSuite/evosuite#18]
Trying to run 
`mvn evosuite:coverage`
on a complex project results in this exception:
```
[ERROR] Failed to execute goal org.evosuite.plugins:evosuite-maven-plugin:1.0.2:coverage (default-cli) on project XYZ: Execution default-cli of goal org.evosuite.plugins:evosuite-maven-plugin:1.0.2:coverage failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.evosuite.plugins:evosuite-maven-plugin:1.0.2:coverage (default-cli) on project XYZ: Execution default-cli of goal org.evosuite.plugins:evosuite-maven-plugin:1.0.2:coverage failed.
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.evosuite.plugins:evosuite-maven-plugin:1.0.2:coverage failed.
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
Caused by: java.lang.NullPointerException
        at org.evosuite.maven.util.ProjectUtils.lambda$getDependencyPathElements$9(ProjectUtils.java:119)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
        at java.util.Iterator.forEachRemaining(Iterator.java:116)
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
        at org.evosuite.maven.util.ProjectUtils.getDependencyPathElements(ProjectUtils.java:122)
        at org.evosuite.maven.CoverageMojo.execute(CoverageMojo.java:112)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        ... 21 more
```

It seems this is caused by dependencies with `<scope>test</scope>`, as soon as I remove the dependency it goes on to find the next pom with a test-scoped dependency and crashes there.
That also might answer the question in line 117 of ProjectUtils with "yes" ;)

**Comment by arcuri82** [comment]
Hi,
  thx. That was an easy fix ;) which is now in 1.0.4-SNAPSHOT (not released). Note however that evosuite:coverage is a non supported feature which is not even documented at 

http://www.evosuite.org/documentation/maven-plugin/

it was implemented as part of the work in progress / experimental Jenkins plugin

cheers

Andrea

**Comment by bseiller** [comment]
I see :)
`mvn evosuite:help`
gave it away, while i was playing around with it.
Thanks for pointing this out to me!
